src/control/server/instance_storage_rpc.go

```diff
@@ -326,18 +387,29 @@ func scanEngineBdevsOverDrpc(ctx context.Context, engine Engine, pbReq *ctlpb.Sc
 		// Populate health if requested.
 		healthUpdated := false
 		if pbReq.Health && c.HealthStats == nil {
-			bhReq := &ctlpb.BioHealthReq{
-				DevUuid:  sd.Uuid,
-				MetaSize: pbReq.MetaSize,
-				RdbSize:  pbReq.RdbSize,
+			bhReq := &ctlpb.BioHealthReq{DevUuid: sd.Uuid}
+			if pbReq.Meta {
+				bhReq.MetaSize = pbReq.MetaSize
+				bhReq.RdbSize = pbReq.RdbSize
+			}
+
+			chReq := ctrlrHealthReq{
+				engine: engine,
+				bhReq:  bhReq,
+				ctrlr:  c,
+			}
+			if !pbReq.Meta {
+				// Add link stats to health only if health requested w/o usage flag.
+				chReq.linkStatsProv = linkStatsProv
 			}
-			upd, err := populateCtrlrHealth(ctx, engine, bhReq, c,
-				pciutils.NewPCIeLinkStatsProvider())
+
+			healthUpdated, err = populateCtrlrHealth(ctx, chReq)
 			if err != nil {
 				return nil, err
 			}
-			healthUpdated = upd
 		}
+		// Used to update health with link stats, now redundant.
+		c.PciCfg = ""
```

Comment:
A bit confused by this. Maybe populateCtrlrHealth should be the one clearing this, if it needs to be cleared.

Reply:
I moved this out of the populate function because we want it to be cleared from here on whether or not it gets used and it didn't seem sensible doing it in 2 places.

Reply:
Understood, although we are still doing it in two places here. I'm OK with it as-is though.